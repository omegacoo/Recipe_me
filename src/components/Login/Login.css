.Login {
    border: 2px solid black;
    border-radius: 15px;
    width: 55vw;
    text-align: center;
    padding: 5px;
    padding-bottom: 15px;
    margin-left: auto;
    margin-right: auto;
    margin-top: 25vh;
}

.Login button {
    margin-top: 15px;
    width: 20vw;
}