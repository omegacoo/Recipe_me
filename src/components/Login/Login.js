import React from 'react';

import './Login.css';

export default class Login extends React.Component {
    state = {
        username: '',
        password: ''
    };

    handleUsernameChange = e => {
        this.setState({
            username: e.target.value
        });
    };

    handlePasswordChange = e => {
        this.setState({
            password: e.target.value
        });
    };

    handleSubmit = e => {
        e.preventDefault();
    };

    render(){
        return(
            <form 
                className='Login'
                onSubmit={this.handleSubmit}
            >
                <label htmlFor='Login_username'>Username: </label>
                <input 
                    className='Login_username'
                    type='text'
                    value={this.state.username}
                    onChange={this.handleUsernameChange}
                /><br></br>
                <label htmlFor='Login_password'>Password: </label>
                <input 
                    className='Login_password'
                    type='password'
                    value={this.state.password}
                    onChange={this.handlePasswordChange}
                />
                <button
                    type='submit'
                >
                    Login
                </button>
            </form>
        );
    };
};