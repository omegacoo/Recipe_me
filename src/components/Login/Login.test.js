import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import Login from './Login';

describe('Login component', () => {
    it('renders without crashing', () => {
        const div = document.createElement('div');

        ReactDOM.render(
            <BrowserRouter>
                <Login />
            </BrowserRouter>, 
            div
        );
        ReactDOM.unmountComponentAtNode(div);
    });

    it('renders UI correctly', () => {
        expect(
        renderer.create(
            <BrowserRouter>
                <Login />
            </BrowserRouter>).toJSON()
        ).toMatchSnapshot();
    });
});