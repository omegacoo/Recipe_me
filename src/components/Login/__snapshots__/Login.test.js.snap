// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login component renders UI correctly 1`] = `
<form
  className="Login"
  onSubmit={[Function]}
>
  <label
    htmlFor="Login_username"
  >
    Username: 
  </label>
  <input
    className="Login_username"
    onChange={[Function]}
    type="text"
    value=""
  />
  <br />
  <label
    htmlFor="Login_password"
  >
    Password: 
  </label>
  <input
    className="Login_password"
    onChange={[Function]}
    type="password"
    value=""
  />
  <button
    type="submit"
  >
    Login
  </button>
</form>
`;
